File: java/BillsApi/src/main/java/com/apilangarena/javabillsapi/application/bills/create/CreateBillCommand.java
package com.apilangarena.javabillsapi.application.bills.create;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.List;

public record CreateBillCommand(
    String billNumber,
    LocalDate issuedAt,
    String customerName,
    String currency,
    BigDecimal tax,
    List<CreateBillLineCommand> lines
) {
}

File: dot-net/BillsApi/Application/Bills/CreateBill/CreateBillCommand.cs
using BillsApi.Application.Abstractions;

namespace BillsApi.Application.Bills.CreateBill;

public sealed record CreateBillLineInput(string Concept, decimal Quantity, decimal UnitAmount);

public sealed record CreateBillCommand(
    string BillNumber,
    DateOnly IssuedAt,
    string CustomerName,
    string Currency,
    decimal Tax,
    IReadOnlyList<CreateBillLineInput> Lines) : ICommand<CreateBillResult>;
